lib/ex_unit/lib/ex_unit/formatter.ex

```diff
@@ -200,25 +202,59 @@ defmodule ExUnit.Formatter do
     formatter.(:error_info, value)
   end
 
-  defp code_multiline(expr, padding) when is_binary(expr) do
-    String.replace(expr, "\n", "\n" <> String.duplicate(" ", padding))
+  defp code_multiline(expr, padding_size) when is_binary(expr) do
+    padding = String.duplicate(" ", padding_size)
+    String.replace(expr, "\n", "\n" <> padding)
   end
 
-  defp code_multiline(expr, padding) do
-    code_multiline(expr |> Macro.to_string, padding)
+  defp code_multiline(expr, padding_size) do
+    code_multiline(Macro.to_string(expr), padding_size)
   end
 
-  defp inspect_multiline(expr, padding, width) do
-    width = if width == :infinity, do: width, else: width - padding
-    expr
-    |> inspect(pretty: true, width: width)
-    |> String.replace("\n", "\n" <> String.duplicate(" ", padding))
+  defp inspect_multiline(expr, padding_size, width) do
+    padding = String.duplicate(" ", padding_size)
+    width = if width == :infinity, do: width, else: width - padding_size
+    inspect(expr, [pretty: true, width: width])
+    |> String.replace("\n", "\n" <> padding)
   end
 
   defp make_into_lines(reasons, padding) do
     padding <> Enum.join(reasons, "\n" <> padding) <> "\n"
   end
 
+  defp format_diff(struct, formatter) do
```

Comment:
What if we make this public as `format_diff(left, right, formatter)` so we can unit test it? :)